### ISSUE TITLE
Error in Priority Update for Prioritized Replay

### ISSUE THREAD

**Issue by qfettes** [higgsfield/RL-Adventure#13]
It looks like you're updating the priorities in the replay buffer according to the _weighted_  and squared TD error.
```
loss  = (q_value - expected_q_value.detach()).pow(2) * weights
prios = loss + 1e-5
replay_buffer.update_priorities(indices, prios.data.cpu().numpy())
```

However, the algorithm in the original paper updates the priority only according to the absolute value of the TD error, which is not weighted. I believe this is a mistake in your implementation

**Comment by mneira10** [comment]
Can confirm:
![image](https://user-images.githubusercontent.com/30906976/65727974-00c6ae80-e07f-11e9-87a4-7280f3b7e0fe.png)
Transition priorities are updated with the **magnitude of the TD error** (lines 11-12).
[Paper for reference](https://arxiv.org/pdf/1511.05952.pdf)

**Comment by sougata-ub** [comment]
Also the loss should be MAE and not MSE
